plan.go

```diff
@@ -262,6 +262,11 @@ func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentTy
 				break
 			}
 		}
+		selectionSetResult = append(selectionSetResult, &ast.Field{
```

Comment:
Could you add a `if !selectionSetHasFieldNamed(selectionSetResult, "__typename") { }` around the append? Our clients send `__typename` through everywhere so this will avoid a lot of duplicate `__typename` selections.

Reply:
I actually propose pulling that method entirely in https://github.com/movio/bramble/pull/94. `selectionSetHasFieldNamed ` is faulty as written because it doesn’t take field aliases into account. If there’s human sensitivity to field duplication in the selection (machines don’t care), I’d propose just running a loop once at the end of extractSelections that does a single-pass uniqueness on alias falling back to name.

Reply:
Ok, that's fair. Let's worry about that at a later stage then (if needed).